### PR TITLE
[cloud-providers][aws] creating natGateway in first non local az

### PR DESCRIPTION
**Pull request by borg-z**
Signed-off-by: borg-z <me@nikolay-z.top>

## Description

This PR updates the NAT Gateway creation process to ensure it is only created in the first public subnet that is not in a local zone.

## Why do we need it, and what problem does it solve?

Local zones have restrictions, such as not supporting NAT Gateway creation. This change ensures NAT Gateways are created in the main availability zones, avoiding these issues.

## Why do we need it in the patch release (if we do)?

Not necessarily.

## What is the expected result?

After applying these changes, NAT Gateways will be created in the first public subnet outside of local zones. This ensures compatibility and prevents creation failures. For a list of local zone restrictions, refer to [AWS Local Zones Features](https://aws.amazon.com/about-aws/global-infrastructure/localzones/features/).

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: candi
type: fix
summary: AWS NAT Gateways are forced to be created in first non-local zone.
impact:
impact_level: default

### CODE REVIEW

**Comment by borg-z** [comment]
Now subnet are created in all availability and local zones, but the NAT Gateway is created in the first availability zone.

```
aws ec2 describe-subnets --filters "Name=vpc-id,Values=vpc-***" --query "Subnets[*].{SubnetId:SubnetId,AvailabilityZone:AvailabilityZone}" --output table --no-cli
-pager 
---------------------------------------------------
|                 DescribeSubnets                 |
+--------------------+----------------------------+
|  AvailabilityZone  |         SubnetId           |
+--------------------+----------------------------+
|  eu-north-1b       |  subnet-0c66ff0a2d78dfecb  |
|  eu-north-1c       |  subnet-09506f411b6315caa  |
|  eu-north-1b       |  subnet-03ce4a00c1fedb79c  |
|  eu-north-1a       |  subnet-0f68fdc31b0aba786  |
|  eu-north-1c       |  subnet-0f7cd5783ce15145b  |
|  eu-north-1-hel-1a |  subnet-0b92690b03a550fd0  |
|  eu-north-1a       |  subnet-0bdec5ee2e8430ab8  |
|  eu-north-1-hel-1a |  subnet-04f3d29cc88500129  |
+--------------------+----------------------------+
```


```
 aws ec2 describe-nat-gateways --filter "Name=vpc-id,Values=vpc-05364a5ebd9852004" --query "NatGateways[*].{NatGatewayId:NatGatewayId,SubnetId:SubnetId}" --output table --no-cli-pager     
-------------------------------------------------------
|                 DescribeNatGateways                 |
+------------------------+----------------------------+
|      NatGatewayId      |         SubnetId           |
+------------------------+----------------------------+
|  nat-0799e7862c6b4083e |  subnet-0f68fdc31b0aba786  |
+------------------------+----------------------------+
```


```
aws ec2 describe-subnets --subnet-ids subnet-0f68fdc31b0aba786 --query "Subnets[*].{SubnetId:SubnetId,AvailabilityZone:AvailabilityZone}" --output table --no-cli-pager

--------------------------------------------------
|                 DescribeSubnets                |
+-------------------+----------------------------+
| AvailabilityZone  |         SubnetId           |
+-------------------+----------------------------+
|  eu-north-1a      |  subnet-0f68fdc31b0aba786  |
+-------------------+----------------------------+

```

**Review by name212 — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/aws'...

**Comment by borg-z** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.27 9994203212 9994203212-1-con-1-27 /sys/deckhouse-oss/install:pr9063 ec2-user@18.198.91.237

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/aws'...

**Review by miklezzzz — APPROVED** [review]

**Comment by apolovov** [comment]
PR summary isn't precise.
I suggest:
```
section: candi
type: fix
summary: AWS NAT Gateways are forced to be created in first non-local zone.
impact: <none>
impact_level: default
```

File: candi/cloud-providers/aws/layouts/with-nat/base-infrastructure/main.tf [hunk 1]
```diff
@@ -31,9 +31,17 @@ module "security-groups" {
...
+  filter {
+    name   = "opt-in-status"
+    values = ["opt-in-not-required"]
+  }
+}
+
 locals {
   az_count    = length(data.aws_availability_zones.available.names)
   subnet_cidr = lookup(var.providerClusterConfiguration, "nodeNetworkCIDR", module.vpc.cidr_block)
+
```
Comment (apolovov) [inline candi/cloud-providers/aws/layouts/with-nat/base-infrastructure/main.tf:44]:
```suggestion
```

**Review by apolovov — CHANGES_REQUESTED** [review]
PR summary isn't precise.
I suggest:
```
section: candi
type: fix
summary: AWS NAT Gateways are forced to be created in first non-local zone.
impact: <none>
impact_level: default
```

**Review by apolovov — APPROVED** [review]

**Review by name212 — APPROVED** [review]

**Pull request merged**